debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnection.java

```diff
@@ -27,33 +14,56 @@
+import org.apache.kafka.connect.errors.ConnectException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
 
 /**
  * A context for a JDBC connection to MySQL.
  *
  * @author Randall Hauch
  */
-public class MySqlJdbcContext implements AutoCloseable {
+public class MySqlConnection extends JdbcConnection {
 
-    protected static final String MYSQL_CONNECTION_URL = "jdbc:mysql://${hostname}:${port}/?useInformationSchema=true&nullCatalogMeansCurrent=false&useSSL=${useSSL}&useUnicode=true&characterEncoding=UTF-8&characterSetResults=UTF-8&zeroDateTimeBehavior=convertToNull";
+//    protected static final String MYSQL_CONNECTION_URL = "jdbc:mysql://${hostname}:${port}/?useInformationSchema=true&nullCatalogMeansCurrent=false&useSSL=${useSSL}&useUnicode=true&characterEncoding=UTF-8&characterSetResults=UTF-8&zeroDateTimeBehavior=convertToNull";
+    protected static final String MYSQL_CONNECTION_URL = "jdbc:mysql://${hostname}:${port}/?useInformationSchema=true&nullCatalogMeansCurrent=false&useSSL=${useSSL}&useUnicode=true&characterEncoding=UTF-8&characterSetResults=UTF-8&zeroDateTimeBehavior=${zeroDateTimeBehavior}";
     protected static final String JDBC_PROPERTY_LEGACY_DATETIME = "useLegacyDatetimeCode";
 
-    protected static ConnectionFactory FACTORY = JdbcConnection.patternBasedFactory(MYSQL_CONNECTION_URL);
+    protected static ConnectionFactory FACTORY = JdbcConnection.patternBasedFactory(
+            MYSQL_CONNECTION_URL,
+            com.mysql.jdbc.Driver.class.getName(),
+            MySqlConnection.class.getClassLoader());
 
-    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    protected static final Logger logger = LoggerFactory.getLogger(MySqlConnection.class);
     protected final Configuration config;
     protected final JdbcConnection jdbc;
-    private final Map<String, String> originalSystemProperties = new HashMap<>();
+    private static boolean DRIVER_MAJOR_VERSION_PRE8;
+    private static final Map<String, String> originalSystemProperties = new HashMap<>();
 
-    public MySqlJdbcContext(Configuration config) {
+    public MySqlConnection(Configuration config) {
+        super(getMysqlConfig(config), FACTORY, MySqlConnection::checkMySqlDriverVersion, MySqlConnection::mySqlConnectionSettings);
+        setupSystemProperties();
         this.config = config; // must be set before most methods are used
 
-        // Set up the JDBC connection without actually connecting, with extra MySQL-specific properties
+        /*// Set up the JDBC connection without actually connecting, with extra MySQL-specific properties
```

Comment:
Hum, why is all this commented out? I'm seeing some commit errors now.

Reply:
This is commented as the code is moved to the 2 different methods. Would remove it.